app/views/fun_page/_canvas.html.erb

```diff
@@ -0,0 +1,28 @@
+<!DOCTYPE html>
+<html>
+<head>
+
+</head>
+<body>
+  <script type="text/javascript">
+    (function(e,t){"use strict";function u(){s&&r.removeChild(s),s=n.createElement("iframe"),s.setAttribute("width","100%"),s.setAttribute("height","100%"),r.appendChild(s)}function f(e){a=e}function l(e){var t=a;e.write("<" + "script>");for(var n=0;n<t.length;n++)e.write("window."+t[n]+" = function(){};");e.write("</" + "script>")}function c(e){u(),s.addEventListener("load",function(){h("load",null)},!1);var t=s.contentWindow.document;t.open(),l(t),t.write(e),t.close()}function h(t,n){var r=i.stringify({data:n,type:t,secret:o});e.parent.postMessage(r,"*")}function p(e,t){var n=s.contentWindow.Error;if(e&&e instanceof Error||e instanceof n)e={message:e.message,stack:e.stack,type:e.type,arguments:e.arguments,__errorType__:String(e.constructor).trim().match(/^function ([^\(\s]+)/)[1]};typeof t=="function"&&(t=String(t)),h("evaljs",{error:e,result:t})}function d(e){var t=null;try{t=s.contentWindow.eval(e)}catch(n){throw p(n,null),n}p(null,t)}function v(){h("html",s.contentWindow.document.documentElement.outerHTML)}window.__webMiniPreview=!0;var n=e.document,r=n.querySelector("body"),i=e.JSON,s,o,a=[],m={load:c,evaljs:d,html:v,disableFunctions:f};e.addEventListener("message",function(e){var t;try{t=i.parse(e.data)}catch(n){return}var r=t.type,s=t.data;if(!t.secret)return;if(!o&&t.type==="handshake")o=t.secret;else{if(t.secret!==o)return;m[r](s)}},!1),e.stuffEmit=function(e,t){h(e,t)}})(window);
+  </script>
+
+  <iframe width="100%" height="100%">
+    <html>
+      <head>
+        <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.7.1/jquery.min.js"></script>
+        <script src="/assets/jquery.expect.js"></script><script src="/assets/web_libs.js?v5"></script>
+        <script type="text/javascript" src="//code.jquery.com/jquery-1.10.2.min.js"></script>
+        <script type="text/javascript" src="http://s3.amazonaws.com/codecademy-content/courses/hour-of-code/js/alphabet.js"></script>
+      </head>
+      <body>
+        <canvas id="myCanvas" height="500" width="1000"></canvas>
+        <script type="text/javascript" src="http://s3.amazonaws.com/codecademy-content/courses/hour-of-code/js/bubbles.js"></script>
+        <%= javascript_include_tag 'bubbles_main.js'%>
```

Comment:
why not load the application.js file here instead and require all of the custom files there?